### PR TITLE
broadcast model in chunks

### PR DESCRIPTION
**Pull request by rishic3**

### CODE REVIEW

**Comment by leewyang** [comment]
build

**Review by leewyang — COMMENTED** [review]
LGTM, but wondering if it's possible to add a simple test for this feature (e.g. by setting the `BROADCAST_LIMIT` to a small value).

**Comment by rishic3** [comment]
build

**Review by leewyang — APPROVED** [review]

File: python/src/spark_rapids_ml/umap.py [hunk 1]
```diff
@@ -675,14 +698,14 @@ def _construct_umap() -> CumlT:
             from .utils import cudf_to_cuml_array
 
             embedding = (
-                self.embedding_
-                if isinstance(self.embedding_, np.ndarray)
-                else self.embedding_.value
+                self.embedding_[0].value
+                if len(self.embedding_) == 1
+                else np.concatenate([chunk.value for chunk in self.embedding_])
```
Comment (wbo4958) [inline python/src/spark_rapids_ml/umap.py:703]: Sorry for the late review, Please don't use any "self" in the function running on the executor side, which will pack everything of "self" and send to executor from driver.

**Pull request merged**